src/app/[locale]/teacher/class-based/[id]/assignments/page.tsx

```
@@ -0,0 +1,44 @@
+    const params = useParams();
+    if (typeof params.id !== 'string' || isNaN(Number(params.id))) {
+        return <div>Invalid classId, please try again.</div>;
+    }
+    const classId = Number(params.id);
+    
+    // temporarily fetch data from server, will change to getting data from global state, avoiding refetching data
+    const {
+        data: myClass,
+        error: myClassError,
+        loading: myClassLoading,
+    } = useFetch<IClass>(() => teacherClassService.getClassById(classId));
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)